x-pack/plugins/endpoint/public/applications/endpoint/store/managing/middleware.ts

```diff
@@ -32,5 +38,13 @@ export const managementMiddlewareFactory: MiddlewareFactory<ManagementListState>
         payload: response,
       });
     }
+    if (action.type === 'userChangedUrl' && hasSelectedHost !== undefined) {
+      const { selected_host: selectedHost } = uiQueryParams(getState());
+      const response = await coreStart.http.get(`/api/endpoint/metadata/${selectedHost}`);
```

Comment:
since the `selectedHost` value is coming from the URL and thus can be manipulated, make sure to test with an invalid host id. The API currently returns a `404` if the endpoint metadata is not found. That will then (I assume) have some implications for the UI